src/services/api/contract_class.rs

```diff
@@ -93,3 +112,100 @@ impl ContractClass {
         Ok(())
     }
 }
+
+// -------------------
+//  Helper Functions
+// -------------------
+
+impl TryFrom<PathBuf> for ContractClass {
+    type Error = io::Error;
+
+    fn try_from(path: PathBuf) -> io::Result<Self> {
+        let file = File::open(path)?;
+        let reader = BufReader::new(file);
+        let raw_contract_class: starknet_api::state::ContractClass =
+            serde_json::from_reader(reader)?;
+
+        let contract_class = ContractClass::from(raw_contract_class);
+        Ok(contract_class)
+    }
+}
+
+fn convert_entry_points(
+    entry_points: HashMap<starknet_api::state::EntryPointType, Vec<EntryPoint>>,
+) -> HashMap<EntryPointType, Vec<ContractEntryPoint>> {
+    let mut converted_entries: HashMap<EntryPointType, Vec<ContractEntryPoint>> = HashMap::new();
+    for (entry_type, vec) in entry_points {
+        let en_type = match entry_type {
+            starknet_api::state::EntryPointType::Constructor => {
+                crate::services::api::contract_class::EntryPointType::Constructor
+            }
+            starknet_api::state::EntryPointType::External => {
+                crate::services::api::contract_class::EntryPointType::External
+            }
+            starknet_api::state::EntryPointType::L1Handler => {
+                crate::services::api::contract_class::EntryPointType::L1Handler
+            }
+        };
+
+        let contracts_entry_points = vec
+            .into_iter()
+            .map(|e| {
+                let selector = Felt::from_bytes_be(e.selector.0.bytes());
+                let offset = e.offset.0.into();
+                ContractEntryPoint { selector, offset }
+            })
+            .collect::<Vec<ContractEntryPoint>>();
+
+        converted_entries.insert(en_type, contracts_entry_points);
+    }
+
+    converted_entries
+}
+
+fn to_cairo_runner_program(
+    program: &starknet_api::state::Program,
+) -> Result<Program, ProgramError> {
+    let program = program.clone();
+    let identifiers = serde_json::from_value::<HashMap<String, Identifier>>(program.identifiers)?;
+
+    let start = match identifiers.get("__main__.__start__") {
```

Comment:
I'm not sure if it applies here, but the JSON has a `main_scope` field, which is usually set to `__main__`. If it's the same `__main__`, then maybe it should use `main_scope` instead of being hard-coded?

Reply:
i can't say if it is the same or not, mainly because it is a function borrowed from starknet-api, we should check it out.